app/models/api/withdrawal_attributes.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Api
+  class WithdrawalAttributes
+    include ActiveModel::AttributeAssignment
+    include ActiveModel::Attributes
+    include ActiveModel::Model
+
+    include DatesHelper
+
+    validate :withdraw_date_valid
+    validates :withdraw_reasons_details, length: { maximum: 1000 }, allow_blank: true
+    validates :withdraw_reasons_dfe_details, length: { maximum: 1000 }, allow_blank: true
+
+    validates :reasons, presence: true
+    validate :unknown_exclusively
+
+    attr_accessor :trainee
+
+    ATTRIBUTES = %i[
+      reasons
+      withdraw_date
+      withdraw_reasons_details
+      withdraw_reasons_dfe_details
+    ].freeze
+
+    ATTRIBUTES.each do |attr|
+      attribute attr
+    end
+
+    def initialize(trainee:)
+      @trainee = trainee
+      super
+    end
+
+    def save!
+      if valid?
+        trainee.update!(attributes.symbolize_keys.except(:reasons).merge(withdrawal_reasons:))
+        trainee.withdraw!
```

Comment:
What happens if the model throws a validation error?

Reply:
A 500 error i guess, it was lifted from here
https://github.com/DFE-Digital/register-trainee-teachers/blob/adf92bee4cb628e25f1d27d21b2dabeada3f264b/app/forms/withdrawal/confirm_details_form.rb#L26-L35